### DISCUSSION TITLE
Including excerpts of another program output

### DISCUSSION THREAD

**Discussion by beldy-cola** [Textualize/rich#1455] [Q&A]
I am building an application with Rich.

Now I want to launch a 3rd party program in a subprocess and capture its output to show some bits of it in my application. The 3rd party program has ansi colored output which helps a lot with grasping it so I would like to retain. 

Is there a way to print colored excerpts right away using Rich builtins?

Of course I can build a simple parser to convert ANSI codes to Rich formatting but I would like not to do so unless there is no easy way to use Rich tools as I think my code will be really bad and slow.

**Comment by willmcgugan** [comment]
You are in luck. There is an undocumented [AnsiDecoder](https://github.com/willmcgugan/rich/blob/master/rich/ansi.py) class which will convert ansi codes in Rich's Text objects.
